src/test/java/features/ScenarioTest.java

```diff
@@ -20,13 +33,50 @@ protected static DataTable ignoreHeadersRow(DataTable datatable) {
         return new TypeReference<>() {};
     }
 
-    protected static <T> T readRequestBody(HandlerRequestBuilder<Map<String, Object>> requestBuilder, Class<T> clazz)
+    protected static <T> T readRequestBody(HandlerRequestBuilder<Map<String, Object>> requestBuilder,
+                                           Class<T> requestBodyClass)
         throws JsonProcessingException {
         Map<String, Object> bodyMap = requestBuilder.getBody(createRequestBuilderTypeRef());
-        return JsonUtils.objectMapper.convertValue(bodyMap, clazz);
+        return JsonUtils.objectMapper.convertValue(bodyMap, requestBodyClass);
     }
 
     protected static <T> Map<String, Object> prepareRequestBody(T requestObject) {
         return JsonUtils.objectMapper.convertValue(requestObject, createRequestBuilderTypeRef());
     }
+
+    protected <I, O> void handlerSendsRequestAndUpdatesResponse(ApiGatewayHandler<I, O> handler) throws IOException {
+        InputStream request = buildRequestInputStream();
+        ByteArrayOutputStream outputStream = invokeHandlerWithRequest(handler, request);
+        scenarioContext.setRequestResponse(outputStream.toString());
+    }
+
+    protected HandlerRequestBuilder<Map<String, Object>> getRequestBuilder() {
+        return scenarioContext.getRequestBuilder();
+    }
+
+    protected DatabaseServiceImpl getDatabaseService() {
+        return scenarioContext.getDatabaseService();
+    }
+
+    private <I, O> ByteArrayOutputStream invokeHandlerWithRequest(
+        ApiGatewayHandler<I, O> handler,
+        InputStream request) throws IOException {
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        Context context = mock(Context.class);
+        handler.handleRequest(request, outputStream, context);
```

Comment:
Probably just inline the variables

Reply:
only context can be inlined. outputStream in returned